NodeRequest member function instead of v8::Context

mikemorris opened the pull request:
Resurrects #5527, but with a JavaScript shim in front to preserve the callback-passing public API.

/cc @tmpsantos @jfirebaugh @miccolis

jfirebaugh commented:
To bulletproof this for production, we should probably tweak the shim so that the exported `Map.prototype` has all the same properties as before, in case somebody is using  `Map.prototype.load.call(map, ...)` or something like that.

I haven't actually tested it, but the following might work:

```
var Map = mbgl.Map;
var originalConstructor = Map.prototype.constructor;
Map.prototype.constructor = function(options) {
    ...
}
```

mikemorris commented:
Rebased this and updated the shim to export the proper prototype, added tests (which were passing on `master`, failing here before but fixed now) to confirm expected behavior.
